weatherman.py

```diff
@@ -0,0 +1,331 @@
+import os
+from collections import namedtuple
+from datetime import datetime
+import argparse
+
+
+# consts for indexes in WeatherReading tuple
+DATE_INDEX = 0
+MAX_TEMP_INDEX = 1
+MEAN_TEMP_INDEX = 2
+MIN_TEMP_INDEX = 3
+MAX_HUMIDITY_INDEX = 4
+MEAN_HUMIDITY_INDEX = 5
+MIN_HUMIDITY_INDEX = 6
+
+# data structure to hold weather readings for a given day
+WeatherReading = namedtuple('WeatherReading', [
+    'date', 'max_temp', 'mean_temp', 'min_temp', 'max_humidity', 'mean_humidity', 'min_humidity'
+])
+
+
+class DictDataItem:
+    ''' Dictionary data item to store key value pairs '''
```

Comment:
Docstring should me more descriptive and functions should have their own docstrings as well